### PR TITLE
♻️  Refactor Environment Variable Usage (#446)

### PR DESCRIPTION
**Pull request by mistryrn**
### Consent UI
* Removes use of blocking fetch call to set client-side runtime env vars as this is no longer required
* Deletes `/api/ready` Next.js API endpoint since it is no longer used
* Adds separate `ClientAppConfig`, containing only the env vars we wish to expose to client components
* Populates `AppConfigContext` with `ClientAppConfig` instead of the whole `AppConfig` to prevent secrets from being exposed to the client
* Remove unnecessary `serverEnv` prop from `getAppConfig` definition and function calls

## PR Readiness Checklist

- [ ] "Expected Outcome(s)" in ticket have been met
- [x] Ticket number included in PR title
- [x] Connected ticket to PR
- [x] Labels added to PR for service name (`consent-api`, `data-mapper`, etc...), type (`chore`, `documentation`, etc...), status (`draft`, `on-hold`, etc...) **if applicable**
- [x] Manual testing completed
- [x] Builds locally without errors or warnings
- [ ] Tests are updated (if required) and passing
- [ ] PR feedback has been addressed **if applicable**
- [ ] New environment variables added to `.env.schema` files, `README.md`
- [ ] Added copyrights to new files

### CODE REVIEW

File: apps/consent-ui/src/config/clientConfig.ts [hunk 1]
```diff
@@ -17,47 +17,37 @@
...
 };
 
 /**
- * Get environment variables for Client Components (i.e. AppConfigContext)
-
- * This is a workaround to expose runtime env vars to the client. We make
- * a dummy fetch call which forces the code to be run server side, and
- * then access and expose process.env to the client.
+ * Returns environment variables for client components
+ * @returns {ClientAppConfig}
```
Comment (anncatton) [inline apps/consent-ui/src/config/clientConfig.ts:43]: should we add an extra note in the jsdoc description that this config should only be used for vars that can safely be exposed in the client code?

File: apps/consent-ui/src/config/appConfig.ts [hunk 2]
```diff
@@ -42,41 +42,19 @@ export const defaultAppConfig: AppConfig = {
...
-		defaultAppConfig.KEYCLOAK_CLIENT_ID,
-	TOKEN_ENCRYPTION_KEY:
-		serverEnv.TOKEN_ENCRYPTION_KEY ||
-		process.env.TOKEN_ENCRYPTION_KEY ||
-		defaultAppConfig.TOKEN_ENCRYPTION_KEY,
-	TOKEN_MAX_AGE:
-		Number(serverEnv.TOKEN_MAX_AGE) ||
-		Number(process.env.TOKEN_MAX_AGE) ||
-		defaultAppConfig.TOKEN_MAX_AGE,
+const getAppConfig = (): AppConfig => ({
```
Comment (anncatton) [inline apps/consent-ui/src/config/appConfig.ts:48]: can we add a `TODO` here or at the top of the file, that we want to add `server-only` here? To be done in this ticket https://github.com/OHCRN/platform/issues/422

Or, we could as a separate task that is dependent on 422.

**Review by anncatton — COMMENTED** [review]
Looks good! Thanks for organizing this. I just added a couple suggestions for comments/docs

File: apps/consent-ui/src/config/appConfig.ts (hunk 2 above)
Comment (mistryrn) [reply apps/consent-ui/src/config/appConfig.ts:48]: Will do 🙂

File: apps/consent-ui/src/config/appConfig.ts [hunk 3]
```diff
@@ -45,6 +45,7 @@ export const defaultAppConfig: AppConfig = {
  * Returns environment variables for server components
  * @returns {AppConfig}
  */
+// TODO: enforce server-only usage. To be completed as a follow-up to https://github.com/OHCRN/platform/issues/422
```
Comment (anncatton) [inline apps/consent-ui/src/config/appConfig.ts:48]: 👍

File: apps/consent-ui/src/config/clientConfig.ts [hunk 4]
```diff
@@ -40,8 +42,11 @@ export const defaultClientAppConfig: ClientAppConfig = {
 
 /**
  * Returns environment variables for client components
+ *
+ * Note: intended for **SERVER COMPONENTS ONLY**. Access these values from the client using the `useAppConfigContext` hook.
```
Comment (anncatton) [inline apps/consent-ui/src/config/clientConfig.ts:46]: ALL CAPS, everyone gotta listen up

File: apps/consent-ui/src/config/clientConfig.ts [hunk 5]
```diff
@@ -40,8 +42,11 @@ export const defaultClientAppConfig: ClientAppConfig = {
...
  * @returns {ClientAppConfig}
  */
+// TODO: enforce server-only usage. To be completed as a follow-up to https://github.com/OHCRN/platform/issues/422
```
Comment (anncatton) [inline apps/consent-ui/src/config/clientConfig.ts:49]: 👍

**Review by anncatton — APPROVED** [review]
lgtm to me, let's see if @ciaranschutte is on board

**Review by ciaranschutte — APPROVED** [review]

**Pull request merged**